### PR TITLE
PP-5159 Refactor stripe requests

### PR DESCRIPTION
**Pull request by mrlumbu**
Make stripe requests simpler, with more of logic in base class

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java [hunk 1]
```diff
@@ -32,17 +44,59 @@ protected StripeRequest(GatewayAccountEntity gatewayAccount, String idempotencyK
...
+        return new GatewayOrder(orderRequestType(), payload, APPLICATION_FORM_URLENCODED_TYPE);
+    }
+    
+    protected Map<String, String> headers() {
+        return Collections.emptyMap();
+    }
+    
+    protected List<String> expansionFields() {
+        return Collections.emptyList();
+    };
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java]: Can we remove `;` at the end of method ?

File: src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java [hunk 2]
```diff
@@ -32,17 +44,59 @@ protected StripeRequest(GatewayAccountEntity gatewayAccount, String idempotencyK
...
+    
+    protected  String getIdempotencyKeyType() {
+        return orderRequestType().toString();
+    };
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java]: Can we remove ; at the end of method ?

File: src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java [hunk 3]
```diff
@@ -32,17 +44,59 @@ protected StripeRequest(GatewayAccountEntity gatewayAccount, String idempotencyK
...
+
+    protected Map<String, String> params() {
+        return Collections.emptyMap();
+    };
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java]: and here

**Review by kbottla — COMMENTED** [review]
Looks good to me! Just a few very minor comments

File: src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java (hunk 1 above)
Comment (mrlumbu) [reply src/main/java/uk/gov/pay/connector/gateway/stripe/request/StripeRequest.java]: eww weird, dunno where that came from

**Review by kbottla — APPROVED** [review]

**Pull request merged**